ProcessLib/BoundaryCondition/PressureBoundaryConditionLocalAssembler.h

```diff
@@ -0,0 +1,174 @@
+        ShapeMatrixPolicyType<ShapeFunctionDisplacement, GlobalDim>;
+    using GlobalDimVectorType =
+        typename ShapeMatrixPolicyType<ShapeFunctionDisplacement,
+                                       GlobalDim>::GlobalDimVectorType;
+
+    PressureBoundaryConditionLocalAssembler(MeshLib::Element const& e,
+                                            std::size_t const local_matrix_size,
+                                            bool is_axially_symmetric,
+                                            unsigned const integration_order,
+                                            Parameter<double> const& pressure)
+        : _integration_method(integration_order), _pressure(pressure)
+    {
+        _local_rhs.setZero(local_matrix_size);
+
+        unsigned const n_integration_points =
+            _integration_method.getNumberOfPoints();
+
+        _ip_data.reserve(n_integration_points);
+
+        auto const shape_matrices_u =
+            initShapeMatrices<ShapeFunctionDisplacement,
+                              ShapeMatricesTypeDisplacement, IntegrationMethod,
+                              GlobalDim>(e, is_axially_symmetric,
+                                         _integration_method);
+
+        GlobalDimVectorType element_normal(GlobalDim);
+
+        // TODO Extend to rotated 2d meshes and line elements.
+        if (e.getGeomType() == MeshLib::MeshElemType::LINE)
+        {
+            auto v1 = (*e.getNode(1)) - (*e.getNode(0));
+            element_normal[0] = -v1[1];
+            element_normal[1] = v1[0];
+            element_normal.normalize();
+        }
+        else
+        {
+            auto const element_normal_vector =
+                MeshLib::FaceRule::getSurfaceNormal(&e).getNormalizedVector();
+
+            std::copy_n(element_normal_vector.getCoords(), GlobalDim,
+                        element_normal.data());
+        }
+
+        for (unsigned ip = 0; ip < n_integration_points; ip++)
+        {
+            typename ShapeMatricesTypeDisplacement::template MatrixType<
+                GlobalDim, displacement_size>
+                N_u = ShapeMatricesTypeDisplacement::template MatrixType<
+                    GlobalDim, displacement_size>::Zero(GlobalDim,
+                                                        displacement_size);
+            for (int i = 0; i < static_cast<int>(GlobalDim); ++i)
+                N_u.template block<1, displacement_size / GlobalDim>(
+                       i, i * displacement_size / GlobalDim)
+                    .noalias() = shape_matrices_u[ip].N;
+
+            double const integration_weight =
+                _integration_method.getWeightedPoint(ip).getWeight() *
+                shape_matrices_u[ip].integralMeasure *
+                shape_matrices_u[ip].detJ;
+
+            _ip_data.emplace_back(N_u.transpose() * element_normal,
+                                  integration_weight);
+        }
+    }
+
+    void assemble(std::size_t const id,
```

Comment:
Not sure about the member question. The detJ is included in the w factor, see https://github.com/fparisio/ogs/blob/pressure_bc/ProcessLib/BoundaryCondition/PressureBoundaryConditionLocalAssembler.h#L114-L117